samples/apps/autogen-studio/autogenstudio/utils/utils.py

```diff
@@ -410,24 +411,44 @@ def test_model(model: Model):
     return response.choices[0].message.content
 
 
-# summarize_chat_history (messages, model) .. returns a summary of the chat history
+def load_code_execution_config(code_execution_type: CodeExecutionConfigTypes, work_dir: str):
+    """
+    Load the code execution configuration based on the code execution type.
 
+    :param code_execution_type: The code execution type.
+    :param work_dir: The working directory to store code execution files.
+    :return: The code execution configuration.
 
-def summarize_chat_history(task: str, messages: List[Dict[str, str]], model: Model):
+    """
+    work_dir = Path(work_dir)
+    work_dir.mkdir(exist_ok=True)
+    executor = None
+    if code_execution_type == CodeExecutionConfigTypes.local:
+        executor = LocalCommandLineCodeExecutor(work_dir=work_dir)
+    elif code_execution_type == CodeExecutionConfigTypes.docker:
+        executor = DockerCommandLineCodeExecutor(work_dir=work_dir)
```

Comment:
This will create a new docker container for each agent. In the future consider setting up docker container a separate resource (like model) to be assigned to multiple agents.

Reply:
Thanks, this is an important catch! Will do!